allennlp/nn/beam_search.py

```diff
@@ -110,6 +111,11 @@ def search(self,
         # shape: (batch_size, beam_size), (batch_size, beam_size)
         start_top_log_probabilities, start_predicted_classes = \
                 start_class_log_probabilities.topk(self.beam_size)
+        if self.beam_size == 1 and (start_predicted_classes == self._end_index).all():
```

Comment:
Could you elaborate on why this only affects beams of size one? A test demonstrating that it can't happen with a beam size of two would be ideal. Sorry if I'm missing something obvious...

Reply:
So with a beam size > 1, `topk` of course returns multiple actions, so you are guaranteed to have at least one non-empty beam (since only one index returned from `topk` could be the `end_index`). Therefore the bug described in #2486 is avoided with beam size > 1 (i.e. it was only possible for that bug to happen with beam size 1). FWIW I did test that it would fail with a beam size of 2 before implementing the explicit check.

Reply:
Thanks for elaborating!